samcli/commands/init/interactive_init_flow.py

```diff
@@ -224,3 +360,34 @@ def _package_schemas_code(runtime, schemas_api_caller, schema_template_details,
         ) from e
     finally:
         remove(download_location.name)
+
+
+def generate_summary_message(package_type, runtime, base_image, dependency_manager, output_dir, name, app_template):
```

Comment:
You may be able to group some common parts (like initializing the `summary_msg` with the Generating application header for example.

Reply:
The common part are alternating with the uncommon part and since strings are immutable, I don't think this will be possible.

Reply:
Strings being immutable only means that under the hood Python creates a new String when you concatenate two occurrences, which is generally a performance issue only when doing it millions of times.